source/common/protobuf/utility.cc

```diff
@@ -1043,4 +1043,44 @@ void TimestampUtil::systemClockToTimestamp(const SystemTime system_clock_time,
           .time_since_epoch()
           .count()));
 }
+
+void StructUtil::update(ProtobufWkt::Struct& obj, const ProtobufWkt::Struct& with) {
+  auto& obj_fields = *obj.mutable_fields();
+
+  for (auto const& [key, val] : with.fields()) {
+    auto& obj_key = obj_fields[key];
+
+    // If the types are different, the last one wins.
+    const auto val_kind = val.kind_case();
```

Comment:
Nit: `const auto&`

Reply:
kind_case() returns a scalar (https://github.com/protocolbuffers/protobuf/blob/7e95c64dfbd9f725f005c83470d44140f58ce8d0/src/google/protobuf/struct.pb.h#L497 & https://github.com/protocolbuffers/protobuf/blob/7e95c64dfbd9f725f005c83470d44140f58ce8d0/src/google/protobuf/struct.pb.h#L317), we can just copy its value no?